jdbc-connector/src/main/scala/org/sunbird/obsrv/helper/ConnectorHelper.scala

```diff
@@ -0,0 +1,94 @@
+import scala.util.control.Breaks.break
+import scala.util.{Failure, Try}
+
+class ConnectorHelper(config: JDBCConnectorConfig) extends Serializable {
+
+ private final val logger: Logger = LogManager.getLogger(getClass)
+
+  def pullRecords(spark: SparkSession, dsSourceConfig: DatasetSourceConfig, dataset: Dataset, batch: Int): (DataFrame, Long) = {
+    val connectorConfig = dsSourceConfig.connectorConfig
+    val connectorStats = dsSourceConfig.connectorStats
+    val jdbcUrl = s"jdbc:${connectorConfig.jdbcDatabaseType}://${connectorConfig.jdbcHost}:${connectorConfig.jdbcPort}/${connectorConfig.jdbcDatabase}"
+    val offset = batch * connectorConfig.jdbcBatchSize
+    val query: String = getQuery(connectorStats, connectorConfig, dataset, offset)
+    var data: DataFrame = null
+    var batchReadTime: Long = 0
+    var retryCount = 0
+
+    logger.info(s"Started pulling batch ${batch + 1}")
+
+    while (retryCount < config.jdbcConnectionRetry && data == null) {
+      val connectionResult = Try {
+        val readStartTime = System.currentTimeMillis()
+
+        val result =  spark.read.format("jdbc")
+            .option("driver", getDriver(connectorConfig.jdbcDatabaseType))
+            .option("url", jdbcUrl)
+            .option("user", connectorConfig.jdbcUser)
+            .option("password", connectorConfig.jdbcPassword)
+            .option("query", query)
+            .load()
+        batchReadTime += System.currentTimeMillis() - readStartTime
+        result
+      }
+
+      connectionResult match {
+        case Failure(exception) =>
+          logger.error(s"Database Connection failed. Retrying (${retryCount + 1}/${config.jdbcConnectionRetry})...", exception)
+          retryCount += 1
+          DatasetRegistry.updateConnectorDisconnections(dsSourceConfig.datasetId, retryCount)
+          if (retryCount == config.jdbcConnectionRetry) break
+          Thread.sleep(config.jdbcConnectionRetryDelay)
+        case util.Success(df) =>
+          data = df
+      }
+    }
+    (data, batchReadTime)
+  }
+
+  def processRecords(config: JDBCConnectorConfig, kafkaClient: KafkaClient, dataset: DatasetModels.Dataset, batch: Int, data: DataFrame, batchReadTime: Long, dsSourceConfig: DatasetSourceConfig): Unit = {
+    val records = JSONUtil.parseRecords(data)
+    val lastRowTimestamp = data.orderBy(data(dataset.datasetConfig.tsKey).desc).first().getAs[Timestamp](dataset.datasetConfig.tsKey)
+    pushToKafka(config, kafkaClient, dataset, records, dsSourceConfig)
+    DatasetRegistry.updateConnectorStats(dsSourceConfig.datasetId, lastRowTimestamp, data.collect().length)
+    logger.info(s"Batch $batch is processed successfully :: Number of records pulled: ${data.collect().length} :: Avg Batch Read Time: ${batchReadTime/batch}")
+  }
+
+  private def pushToKafka(config: JDBCConnectorConfig, kafkaClient: KafkaClient, dataset: DatasetModels.Dataset, records: List[Map[String, Any]], dsSourceConfig: DatasetSourceConfig): Unit ={
```

Comment:
Use `DataFrame` rather than `List` for `records`. Try to use the below implementation to send the events to Kafka.
https://github.com/Sunbird-Obsrv/sunbird-analytics-core/blob/release-5.1.3/analytics-[core/src/main/scala/org/ekstep/analytics/framework/dispatcher/KafkaDispatcher.scala](https://github.com/Sunbird-Obsrv/sunbird-analytics-core/blob/release-5.1.3/analytics-core/src/main/scala/org/ekstep/analytics/framework/dispatcher/KafkaDispatcher.scala#L55-L79)